src/mapValues.test.ts

```diff
@@ -34,3 +34,51 @@ describe('data last', () => {
     });
   });
 });
+
+describe('mapValues key types', () => {
+  test('interface', () => {
+    mapValues({} as { foo: unknown; bar: unknown }, (_, key) =>
+      expectTypeOf(key).toEqualTypeOf<'foo' | 'bar'>()
+    );
+  });
+
+  test('interface with symbols', () => {
+    const mySymbol = Symbol('mySymbol');
+    mapValues(
+      {} as { [mySymbol]: unknown; foo: unknown; bar: unknown },
+      (_, key) => expectTypeOf(key).toEqualTypeOf<'foo' | 'bar'>()
+    );
+  });
+
+  test('mapped type', () => {
+    mapValues({} as { [K in string]: unknown }, (_, key) => {
+      expectTypeOf(key).toEqualTypeOf<string>();
+    });
+    mapValues({} as { [K in number]: unknown }, (_, key) => {
+      expectTypeOf(key).toEqualTypeOf<`${number}`>();
+    });
+    mapValues({} as { [K in `prefix${string}`]: unknown }, (_, key) => {
+      expectTypeOf(key).toEqualTypeOf<`prefix${string}`>();
+    });
+    mapValues({} as { [K in symbol]: unknown }, (_, key) => {
+      expectTypeOf(key).toEqualTypeOf<never>();
+    });
+  });
```

Comment:
nit: I prefer using a `describe` here and breaking each assert to a separate test so that we get fine-tuned test results with the specific issue. Up to you if you want to change it.